pkg/flag/pairlist.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flag
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+// PairList is a flag value supporting a list of key=value pairs
+// It will optionally validate the supplied keys and / or values
+// It would parse something like "--arg foo=bar --arg baz=bar"
+type PairList struct {
+	Pairs       map[string]string
+	permissible map[string](map[string]struct{}) // map of allowed keys and, optionally, allowed values
+	typeName    string
+}
+
+var _ pflag.Value = (*PairList)(nil)
+
+// EMPTY is a small shortcut for the empty map
+var EMPTY = map[string]string{}
```

Comment:
unfortunately this is not constant, and also being exported here, hence it can be mutated everywhere. If you want to return/pass empty maps, just use `nil`, since it is a valid zero value for maps.